infrastructure/httpapi/handlers/validator_delegation.go

```diff
@@ -0,0 +1,262 @@
+package handlers
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/valyala/fasthttp"
+
+	"github.com/crypto-com/chain-indexing/appinterface/rdb"
+	applogger "github.com/crypto-com/chain-indexing/external/logger"
+	"github.com/crypto-com/chain-indexing/infrastructure/httpapi"
+	validator_delegation_view "github.com/crypto-com/chain-indexing/projection/validator_delegation/view"
+)
+
+type ValidatorDelegation struct {
+	logger applogger.Logger
+
+	accountAddressPrefix   string
+	validatorAddressPrefix string
+	consNodeAddressPrefix  string
+
+	validatorsView           validator_delegation_view.Validators
+	delegationsView          validator_delegation_view.Delegations
+	unbondingDelegationsView validator_delegation_view.UnbondingDelegations
+	redelegationsView        validator_delegation_view.Redelegations
+}
+
+func NewValidatorDelegation(
+	logger applogger.Logger,
+	accountAddressPrefix string,
+	validatorAddressPrefix string,
+	consNodeAddressPrefix string,
+	rdbHandle *rdb.Handle,
+) *ValidatorDelegation {
+	return &ValidatorDelegation{
+		logger.WithFields(applogger.LogFields{
+			"module": "ValidatorDelegationHandler",
+		}),
+
+		accountAddressPrefix,
+		validatorAddressPrefix,
+		consNodeAddressPrefix,
+
+		validator_delegation_view.NewValidatorsView(rdbHandle),
+		validator_delegation_view.NewDelegationsView(rdbHandle),
+		validator_delegation_view.NewUnbondingDelegationsView(rdbHandle),
+		validator_delegation_view.NewRedelegationsView(rdbHandle),
+	}
+}
+
+func (handler *ValidatorDelegation) FindValidatorByAddress(ctx *fasthttp.RequestCtx) {
+
+	height, err := parseInputHeight(ctx)
+	if err != nil {
+		httpapi.BadRequest(ctx, fmt.Errorf("error parsing input height: %v", err))
+		return
+	}
+
+	validatorAddress, ok := ctx.UserValue("address").(string)
+	if !ok {
+		httpapi.BadRequest(ctx, errors.New("error parsing input address"))
+		return
+	}
+
+	var validator validator_delegation_view.ValidatorRow
+	var found bool
+
+	if strings.HasPrefix(validatorAddress, handler.consNodeAddressPrefix) {
+
+		validator, found, err = handler.validatorsView.FindByConsensusNodeAddr(validatorAddress, height)
+		if err != nil {
+			handler.logger.Errorf("error finding validator by ConsensusNodeAddress: %v", err)
+			httpapi.InternalServerError(ctx)
+			return
+		}
+		if !found {
+			httpapi.NotFound(ctx)
+			return
+		}
+
+	} else if strings.HasPrefix(validatorAddress, handler.validatorAddressPrefix) {
+
+		validator, found, err = handler.validatorsView.FindByOperatorAddr(validatorAddress, height)
+		if err != nil {
+			handler.logger.Errorf("error finding validaotr by OperatorAddress: %v", err)
+			httpapi.InternalServerError(ctx)
+			return
+		}
+		if !found {
+			httpapi.NotFound(ctx)
+			return
+		}
+
+	} else {
+		httpapi.BadRequest(ctx, errors.New("invalid address"))
+		return
+	}
+
+	httpapi.Success(ctx, validator)
+}
+
+func (handler *ValidatorDelegation) ListValidator(ctx *fasthttp.RequestCtx) {
```

Comment:
may i know what is the diff between this `ListValidator` and the `List` in `Validator` projection view?